src/text/text.cpp

```diff
@@ -773,22 +773,25 @@ std::string prettify_currency(float num) {
 		"%.0fZ \xA4"
 	};
 	char buffer[200] = { 0 };
-	double dval = double(num);
+	double dval = std::abs(double(num));
 	for(size_t i = std::extent_v<decltype(mag)>; i-- > 0;) {
-		if(std::abs(dval) >= mag[i]) {
-			auto reduced = num / mag[i];
+		if(dval >= mag[i]) {
+			auto reduced = dval / mag[i];
 			if(reduced < 10.0) {
```

Comment:
this is ugly. Instead of writing to one after the start of the buffer, just test abs(reduced)